migration-engine/connectors/sql-migration-connector/src/sql_schema_calculator/datamodel_helpers.rs

```diff
@@ -228,22 +224,38 @@ impl<'a> RelationFieldRef<'a> {
             })
     }
 
-    pub(crate) fn referencing_columns<'b>(&'b self) -> impl Iterator<Item = &'a str> {
-        self.field.data_source_fields().iter().map(|field| field.name.as_str())
+    pub(crate) fn referencing_columns<'b>(&'b self) -> Vec<String> {
```

Comment:
I was unable to keep the `impl Iter` interface, the lifetimes simply didn't add up. If you have a better solution I'm all ears.

Reply:
It's not important at all but I was curious to see if that was possible. Apparently this works: https://github.com/prisma/prisma-engines/pull/686

Reply:
I tried without `move`... This was one of the least obvious things / errors I encountered with rust so far.

Reply:
The interaction of borrowing/lifetimes and closures is something that took me really long to really grasp, and yeah the errors are bad.